app/controllers/lectures_controller.rb

```diff
@@ -16,6 +16,7 @@ def show
     responses = Response.where(lecture: @lecture, user: current_user)
     @emoji_map = responses.map { |res| [res.answer, (res.is_correct? ? '✅' : '✖')]}.to_h
     @livelecture_uri = URI.encode "/livelecture/show?lecture=#{params[:id]}"
+    @course = Course.find_by(id: params[:course_id])
```

Comment:
For this @course, could you use something like where the @lecture is populated? Same for the registration controller